src/request.ts

```diff
@@ -0,0 +1,48 @@
+import got from 'got'
+import qs from 'query-string'
+
+type Params = Record<string, string>
+
+interface RequestOptions {
+  method?: string,
+  body?: object,
+}
+
+const ROOT = 'https://api.geni.us'
+
+const HEADERS = {
+  Accept: 'application/json',
+  'X-Api-Key': undefined as string,
+  'X-Api-Secret': undefined as string,
+}
+
+const setApiKeys = (key: string, secret: string) => {
+  HEADERS['X-Api-Key'] = key
+  HEADERS['X-Api-Secret'] = secret
+}
+
+const makeRequest = (path: string, options?: RequestOptions) => {
+  const url = ROOT + path
+  return got(url, {
+    headers: HEADERS,
+    json: true,
```

Comment:
The biggest change is that we now use `got` -- it automatically handles network errors and retries for us :)